plenario/api/point.py

```diff
@@ -1,552 +1,176 @@
+def dataset_fields(dataset_name):
+    request_args = request.args.to_dict()
+    request_args['dataset_name'] = dataset_name
+    validated_args = validate(DatasetRequiredValidator, request_args)
+    if validated_args.errors:
+        return bad_request(validated_args.errors)
+    return _meta(validated_args)
+
+
+@cache.cached(timeout=CACHE_TIMEOUT, key_prefix=make_cache_key)
+@crossdomain(origin="*")
+def meta():
+    validated_args = validate(NoDefaultDatesValidator, request.args.to_dict())
+    if validated_args.errors:
+        return bad_request(validated_args.errors)
+    return _meta(validated_args)
+
+
+# ============
+# _route logic
+# ============
+
+def _timeseries(args):
+
+    geom = args.data['geom']
+    dataset = args.data.get('dataset')
+    table_names = args.data['dataset_name__in']
+    start_date = args.data['obs_date__ge']
+    end_date = args.data['obs_date__le']
+    agg = args.data['agg']
+
+    # if a single dataset was provided, it's the only thing we need to consider
+    if dataset is not None:
+        table_names = [dataset.name]
+        # for the query's meta information, so that it doesn't show the index
+        del args.data['dataset_name__in']
+
+    # remove table names which wouldn't return anything for the query, given
+    # the time and geom constraints
     try:
         table_names = MetaTable.narrow_candidates(table_names, start_date, end_date, geom)
     except Exception as e:
         msg = 'Failed to gather candidate tables.'
         return internal_error(msg, e)
 
+    # If there aren't any table names, it causes an error down the code. Better
+    # to return and inform them that the request wouldn't have found anything.
+    if not table_names:
+        return bad_request("Your request doesn't return any results. Try "
+                           "adjusting your time constraint or location "
+                           "parameters.")
+
     try:
-        panel = MetaTable.timeseries_all(table_names=table_names,
-                                         agg_unit=agg,
-                                         start=start_date,
-                                         end=end_date,
-                                         geom=geom)
+        panel = MetaTable.timeseries_all(
+            table_names, agg, start_date, end_date, geom
+        )
     except Exception as e:
         msg = 'Failed to construct timeseries.'
         return internal_error(msg, e)
 
     panel = MetaTable.attach_metadata(panel)
-    resp = json_response_base(validator, panel)
+    resp = json_response_base(args, panel, args.data)
 
-    datatype = validator.vals['data_type']
+    datatype = args.data['data_type']
     if datatype == 'json':
-        resp = make_response(json.dumps(resp, default=dthandler), 200)
+        resp = make_response(json.dumps(resp, default=unknown_object_json_handler), 200)
         resp.headers['Content-Type'] = 'application/json'
     elif datatype == 'csv':
```

Comment:
Any way to format the CSV from the dictionary response instead of doing this whole separate thing?